File: scripts/koruh/blocks/expvoid.js
const expvoid = extendContent(Block, "exp-void", {
    hasExp(){
        return true;
    },
    expCapacityf(){
        return 99999999;//heh
    }
});
expvoid.update = true;
expvoid.solid = true;
expvoid.buildType = () => extend(Building, {
    totalExp(){
        return 0;
    },
    consumesOrb(){
        return this.enabled;
    },
    getOrbMuitiplier(){
        return 1;
    },
    incExp(amount){
        return;
    }
});

File: main/src/unity/world/blocks/storage/ExpVoid.java
package unity.world.blocks.storage;

import mindustry.gen.Building;
import mindustry.world.Block;
import unity.world.blocks.*;

public class ExpVoid extends Block implements ExpBlockBase{
    public ExpVoid(String name){
        super(name);
        update = solid = true;
    }

    @Override
    public int expCapaciry(){
        return 99999999;
    }

    public class ExpVoidBuild extends Building implements ExpBuildBase{
        @Override
        public float getMaxExp(){
            return 0f;
        }

        @Override
        public float totalExp(){
            return 0f;
        }

        @Override
        public float expf(){
            return 0f;
        }

        @Override
        public void setExp(float a){}

        @Override
        public boolean consumesOrb(){
            return enabled;
        }

        @Override
        public void incExp(float a){}
    }
}
